### DISCUSSION TITLE
Pull request template: Improving template with more details (checklist or other guidance)

### DISCUSSION THREAD

**Discussion by lmbr-pip** [o3de/o3de#10280] [Ideas]
In https://github.com/o3de/o3de/pull/9912, we delivered a very simple Pull Request (PR) template: https://github.com/o3de/o3de/blob/development/.github/pull_request_template.md

There was a lot of good discussion about how to improve what was being asked, for reference the original PR had this template:

## PR Checklist
_Please put an `x` in the boxes that apply. Maintainers can and will provide guidance as to what should be checked and you can update this checklist after the PR is created._
_Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository._

- [ ] PR includes unit or automation tests. _If not please consider adding tests._
- [ ] Has User facing changes. _If so please consider adding screenshots and asking for [sig/ui-ux](https://github.com/o3de/sig-ui-ux) reviewers._
- [ ] Requires documentation. _If so please consider adding links to documentation issues in [o3de.org](https://github.com/o3de/o3de.org)._
- [ ] Breaking or Impactful change. _If so please consider sending out change notification in advance to #impactful-changes channel on O3DE Discord._

I would like to continue the discussion here so we can arrive at a *high* value checklist (or set of instructions) which do not create a significant hurdle that causes PR abandonment.

Would like to ensure that the guidance:
* Captures information quickly from contributor so maintainers don't have to chase it.
* Provides clear instructions to maintainers so they can easily address any expectations of contributions up-front.
* Is not too complex or burdensome; has to be clear to new contributors and esp non-native English speakers.

Expectation of this discussion:
* We can arrive a small list of details to ask
* Someone makes a PR for it
* We advertise PR widely ie in TSC, sig-all and ensure as many voices are heard
* PR is contributed and we can continue to monitor for feedback and quickly respond to changes.

**Comment by OBWANDO** [comment]
Having the layout as reminder points that can be confirmed is a good approach where someone may not check the box, but at least we know what state it is in. It also does not need to be just for the person submitting the PR, we can also include a checklist for the maintainer as well like "Feature graph updated with changes. If so, please consider updating the appropriate feature in https://github.com/o3de/community/tree/main/features"

I would like to suggest an universal checklist of generic things (like we have here), a Maintainer's checklist (1-2 items), and any checklist specific to a sig.

Something like : 
## PR Checklist
_Please put an `x` in the boxes that apply. Maintainers can and will provide guidance as to what should be checked and you can update this checklist after the PR is created._
_Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository._

#### PR Contribution Checklist:
- [ ] PR includes unit or automation tests. _If not please consider adding tests._
- [ ] Has User facing changes. _If so please consider adding screenshots and asking for [sig/ui-ux](https://github.com/o3de/sig-ui-ux) reviewers._
- [ ] Requires documentation. _If so please consider adding links to documentation issues in [o3de.org](https://github.com/o3de/o3de.org)._
- [ ] Breaking or Impactful change. _If so please consider sending out a change notification in advance to #impactful-changes channel on O3DE Discord._

#### Maintainer Checklist:
- [ ] Feature graph updated with changes. _If so, please consider updating the appropriate feature in [Feature List](https://github.com/o3de/community/tree/main/features)

#### SIG-UI-UX
- [ ] Feature requires diversity review. _If so. please consider including a diversity committee maintainer 


This will allow the basic template to always be the same, and also allow SIGs to have specific elements to guide contributors. These binary "check offs" are very simple and should not be burdensome to the contibutor.

**Comment by Bindless-Chicken** [comment]
I think the goals you've set @lmbr-pip are spot on.
I also really like the list you've put together, I would however add a link to a doc page (that we may have to write) for what we expect from a test and how to write them.
I also like the idea from @OBWANDO to have a reviewer checklist, this could help us track the state of review, and make sure the right questions are asked (and not twice), same for the sig section. Could we consider having a bot comment these? That way we could restrict what info shows up depending on the folders that were touched. It would be pretty confusing to have a bunch of sections (one per sig?) with most of them not applying.
